### PR TITLE
fix: avoid ambiguous messages about whether correspondence can or cannot be guaranteed

### PR DESCRIPTION
**Pull request by jonasbrunvoll**
Closes https://github.com/AtB-AS/kundevendt/issues/17624

### What is wrong, and what is expected behavior?

A user has asked us a question regarding ambiguous messages. The messages at the top of the app say that correspondence cannot be guaranteed, while the message a little further down says that there is correspondence between 340 and 341 (picture 1). This is confusing. 

### Proposed solution
Remove the last sentence "Kan ikke garantere korrespondanse.", if the message box "Korrespondanse mellom 340 og 341 på Volda sjukehus" appears.   

### Illustration Solution
<details>
<Summary>screenshots/video</Summary>

Image

Image

</details>

### CODE REVIEW

**Review by gorandalum — CHANGES_REQUESTED** [review]
I think this will return `false` whenever two legs contains a correspondence between them, even though there might be a non-corresponding short-wait-time between some other legs?

**Comment by jonasbrunvoll** [comment]
> If there is a non-corresponding short wait time between two legs of the trip, then this method should return true. I think this now may be masked by some other corresponding legs, which overrides the answer to false.

`hasGuaranteedCorrespondence(filteredLegs)`, uses the same conditions as the component 'TripSection.tsx, line 352', to determine whether it should return true or false. As I understand it, correspondence is guaranteed if this box is displayed, even when shorWaitTime kicks in. Or maybe this only applies to FRAM?

File: src/travel-details-screens/__tests__/has_short_wait_time_and_not_guaranteed_correspondance.test.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        expectedEndTime: '2024-10-31T15:05:00Z',
+      },
+      {
+        expectedStartTime: '2024-10-31T15:6:00Z',
+        expectedEndTime: '2024-10-31T15:20:00Z',
+      },
+    ] as Leg[];
+
+    expect(hasShortWaitTimeAndNotGuaranteedCorrespondence(legs)).toBe(s7);
+  });
```
Comment (gorandalum) [inline src/travel-details-screens/__tests__/has_short_wait_time_and_not_guaranteed_correspondance.test.ts:108]: The test is intended to test that even if there is no wait time between bus and walking (first and second leg in this example), the function returns false. To actually test that the last leg should start something like 15:10.

As the test input legs are written now with the end time of the walking leg at 15:05 and the start time of the last bus leg is 15:06, the function should actually return true. So there is a error in the implementation somewhere.

**Review by gorandalum — CHANGES_REQUESTED** [review]

File: src/travel-details-screens/__tests__/has_short_wait_time_and_not_guaranteed_correspondance.test.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
+import {hasShortWaitTimeAndNotGuaranteedCorrespondence} from '../utils';
+import {Leg} from '@atb/api/types/trips';
+
+describe('hasShortWaitTimeAndNotGuaranteedCorrespondence', () => {
+  const s1 = false;
```
[the remark this answers is not in the document]
Comment (jonasbrunvoll) [reply src/travel-details-screens/__tests__/has_short_wait_time_and_not_guaranteed_correspondance.test.ts:5]: Okey, I´ll change the tests to make them more readable👍

File: src/travel-details-screens/__tests__/has_short_wait_time_and_not_guaranteed_correspondance.test.ts [hunk 3]
```diff
@@ -115,17 +127,119 @@ describe('hasShortWaitTimeAndNotGuaranteedCorrespondence', () => {
...
+      {
+        expectedStartTime: '2024-10-31T15:21:00Z',
+        expectedEndTime: '2024-10-31T15:30:00Z',
+      },
+    ] as unknown as Leg[];
+
+    expect(hasShortWaitTimeAndNotGuaranteedCorrespondence(legs)).toBe(true);
+  });
+
+  it('should be true because of short wait time between third and forth leg.', () => {
```
Comment (gorandalum) [inline src/travel-details-screens/__tests__/has_short_wait_time_and_not_guaranteed_correspondance.test.ts:219]: Shouldn't have two tests with the same name

File: src/travel-details-screens/__tests__/has_short_wait_time_and_not_guaranteed_correspondance.test.ts [hunk 4]
```diff
@@ -115,17 +127,119 @@ describe('hasShortWaitTimeAndNotGuaranteedCorrespondence', () => {
...
+        expectedEndTime: '2024-10-31T15:00:00Z',
+      },
+      {
+        expectedStartTime: '2024-10-31T15:00:00Z',
+        expectedEndTime: '2024-10-31T15:05:00Z',
+      },
+    ] as Leg[];
+
+    expect(hasShortWaitTimeAndNotGuaranteedCorrespondence(legs)).toBe(true);
+  });
```
Comment (gorandalum) [inline src/travel-details-screens/__tests__/has_short_wait_time_and_not_guaranteed_correspondance.test.ts:191]: I think this one is wrong, and the test should be 'should be false when short wait time after first foot leg' or something like it. 

When there is just walking as the first leg, the journey planner algorithm always make it be a the bus stop "just in time". This means that with this logic, we would get the short wait time message on every trip starting with a foot leg.

**Review by gorandalum — CHANGES_REQUESTED** [review]

File: src/travel-details-screens/__tests__/has_short_wait_time_and_not_guaranteed_correspondance.test.ts (hunk 3 above)
Comment (jonasbrunvoll) [reply src/travel-details-screens/__tests__/has_short_wait_time_and_not_guaranteed_correspondance.test.ts:219]: Okey, i´ll change the name👍

File: src/travel-details-screens/__tests__/has_short_wait_time_and_not_guaranteed_correspondance.test.ts (hunk 4 above)
Comment (jonasbrunvoll) [reply src/travel-details-screens/__tests__/has_short_wait_time_and_not_guaranteed_correspondance.test.ts:191]: Okey, I´ll update the logic👍

**Review by gorandalum — APPROVED** [review]
Great! :+1:

**Pull request merged**